src/pharmpy/tools/run.py

```diff
@@ -857,22 +856,82 @@ def _get_delta(model):
+
+    def __le__(self, value):
+        return all(e <= value for e in self.x)
+
+    def __ge__(self, value):
+        return all(e >= value for e in self.x)
+
+    def __gt__(self, value):
+        return all(e > value for e in self.x)
+
+
+def _strictness_filtering(res: ModelfitResults, statement: str) -> bool:
+    """Takes a model and a statement as input and returns True/False
+    if the evaluation of the statement is True/False.
+    """
+    if res is None or np.isnan(res.ofv):
         return False
-    if not res.minimization_successful:
-        if errors_allowed:
-            if res.termination_cause not in errors_allowed:
-                return False
-            if np.isnan(res.significant_digits):
-                return False
+    if statement is not None:
+        allowed_args = [
+            'minimization_successful',
+            'rounding_errors',
+            'sigdigs',
+            'maxevals_exceeded',
+            'RSE',
+            'condition_number',
+            'final_zero_gradient',
+            'estimate_near_boundary',
+        ]
+        unwanted_args = ['and', 'or', 'not']
+        find_all_words = re.findall(r'[^\d\W]+', statement)
+        args_in_statement = [w for w in find_all_words if w not in unwanted_args]
+
+        # Check that only allowed arguments are in the statement
+        if not all(map(lambda x: x in allowed_args, args_in_statement)):
+            raise ValueError(
+                f'Some expressions were not correct. Valid arguments are: {allowed_args}'
+            )
         else:
-            return False
-    return True
+            minimization_successful = res.minimization_successful
+            rounding_errors = res.termination_cause == "rounding_errors"
+            maxevals_exceeded = res.termination_cause == "maxevals_exceeded"
+            sigdigs = ArrayEvaluator([res.significant_digits])
+            # condition_number = ArrayEvaluator([condition_number]) FIXME: add
+            # final_zero_gradient FIXME: add
+            # estimate_near_boundary FIXME: add
+            if "RSE" in args_in_statement:
+                RSE = ArrayEvaluator(res.relative_standard_errors)
+            else:
+                RSE = None
+
+        # dictionary needed because otherwise unused variables will cause lint error
+        return eval(
+            statement,
+            {
+                'minimization_successful': minimization_successful,
+                'rounding_errors': rounding_errors,
+                'maxevals_exceeded': maxevals_exceeded,
+                'sigdigs': sigdigs,
+                'RSE': RSE,
+            },
+        )
+    else:
+        return True
 
 
-def _get_rankval(model, errors_allowed, rank_type, bic_type, **kwargs):
-    if not _is_valid_res(model.modelfit_results, errors_allowed):
+def _get_rankval(model, strictness, rank_type, bic_type, **kwargs):
+    if not _strictness_filtering(model.modelfit_results, strictness):
```

Comment:
I think with this statement that the name is not super clear, it is a private function so the name can be left as is, but I guess what the function does is to check if a model (or a result) fulfills a strictness criteria rather than filter. To me it would be a bit clearer if it had `is` in the name since it returns a boolean.